### DISCUSSION TITLE
Should text resource be stored in the unmodifiable list?

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1233] [Q&A] [answered]
Hi:

Recently we got a problem that
if we use FXGL.text("....txt"); to load the text resources
then the returned list are modifiable
and if developer change the list then use the FXGL.text method again to load the resources
then it will read the modified list from the cache
and it may cause the unexpected results
so is it a good idea to make text cache list unmodifiable?
like return List.copyOf(originalTextList) for the AssetLoader.load method?

thx

**Comment by AlmasB** [comment, marked as answer]
You are correct, it is a bug, should be fixed as part of #977
